src/pages/home/components/form-guest/form-guest.tsx

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable spaced-comment */
+/* eslint-disable no-restricted-globals */
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable import/no-extraneous-dependencies */
+import * as z from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Cookies } from "react-cookie";
+import { Link } from "react-router-dom";
+import { AxiosError } from "axios";
+import { useState } from "react";
+import styles from "../../home.module.css";
+import { api } from "../../../../api";
+
+interface FormProps {
+  setLoginOrGuest: (typeForm: "login" | "guest") => void;
+}
+
+interface LogimGuestProps {
+  name: string;
+  room: string;
+}
+
+export function FormGuest({ setLoginOrGuest }: FormProps) {
+  const [serverError, setServerError] = useState<string>("");
+
+  const cookies = new Cookies();
+
+  //schema form
```

Comment:
Acredito que essa parte fique melhor fora do componente. O Zod criar um tipo estático com base no seu schema, porém uma nova instância desse tipo vai ser criada cada vez que o componente for re-renderizado.